src/tracks/playabletrack/wavetrack/ui/LowlitClipButton.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include "ClipButtonId.h"
+#include "UIHandle.h"
+#include "WaveTrack.h"
+
+class wxRect;
+class wxPoint;
+
+template <ClipButtonId id> struct ClipButtonTag
+{
+};
+
+struct ClipButtonDrawingArgs
+{
+   const wxRect& rect;
+   const ClipInterface& clip;
+   wxDC& dc;
+};
+
+namespace ClipButtonSpecializations
+{
+bool NeedsDrawing(ClipButtonTag<ClipButtonId::Overflow>, const ClipInterface&);
+bool NeedsDrawing(ClipButtonTag<ClipButtonId::Speed>, const ClipInterface&);
+bool NeedsDrawing(ClipButtonTag<ClipButtonId::Pitch>, const ClipInterface&);
+
+void DrawOnClip(ClipButtonTag<ClipButtonId::Overflow>, ClipButtonDrawingArgs&);
+void DrawOnClip(ClipButtonTag<ClipButtonId::Speed>, ClipButtonDrawingArgs&);
+void DrawOnClip(ClipButtonTag<ClipButtonId::Pitch>, ClipButtonDrawingArgs&);
+
+int GetWidth(ClipButtonTag<ClipButtonId::Overflow>);
+int GetWidth(ClipButtonTag<ClipButtonId::Speed>);
+int GetWidth(ClipButtonTag<ClipButtonId::Pitch>);
+} // namespace ClipButtonSpecializations
+
+namespace LowlitClipButton
+{
+struct RectangleArgs
```

Comment:
:eyes:
https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/avoid-const-or-ref-data-members.html
Not the case in that particular set of arguments, but I don't like the idea of wrapping arguments into a structure to shorten the list...

Reply:
First time I read something like this. I do see advantages.